### ISSUE TITLE
On MACLR algorithm: How can clustering be applied to the training process of step 1?

### ISSUE THREAD

**Issue by YXWisaboy** [amzn/pecos#137]
I found that you apply the clustering results to the loss function, but after torch.eq transformation, only a diagonal matrix will be output

```
def loss_function_reg(label_emb, inst_emb, inst_emb_aug, reg_emb, labels, accelerator):
	assert label_emb.shape[0] == inst_emb.shape[0], "{} is not equal to {}".format(label_emb.shape[0], inst_emb.shape[0])
	assert label_emb.shape[1] == inst_emb.shape[1]

	all_label_emb = mpu_utils.AllgatherFromDataParallelRegion.apply(label_emb)
	all_inst_emb = mpu_utils.AllgatherFromDataParallelRegion.apply(inst_emb)
	all_inst_emb_aug = mpu_utils.AllgatherFromDataParallelRegion.apply(inst_emb_aug)
	all_reg_emb = mpu_utils.AllgatherFromDataParallelRegion.apply(reg_emb)

	labels = labels.contiguous().view(-1, 1)
	all_labels = accelerator.gather(labels)
	num_inst = all_label_emb.shape[0]
	num_reg = all_reg_emb.shape[0]

	mask = torch.eq(all_labels, all_labels.transpose(0, 1)).float()
```

So how can the clustering results be applied to the training process? Is there a code reference？
Hope to get your reply！

**Comment by YXWisaboy** [comment]
Oh, I know how the clustering results work. That line of code won't get the diagonal matrix when two data correspond to a cluster center.
